src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -4412,6 +4412,14 @@ int RGWRados::fetch_remote_obj(RGWObjectCtx& obj_ctx,
     encode(trace, bl);
     cb.get_attrs()[RGW_ATTR_OBJ_REPLICATION_TRACE] = std::move(bl);
   }
+  {
+    // add x-amz-replicated-at
+    auto& bl = cb.get_attrs()[RGW_ATTR_OBJ_REPLICATION_TIMESTAMP];
+    bl.clear();
+    std::string replicated_time = ceph::to_iso_8601_no_separators(ceph::real_clock::now(),
+                                                                  ceph::iso_8601_format::YMDhmsn);
+    bl.append(replicated_time);
```

Comment:
probably better to just store this timestamp in the xattr, and do the iso8601 formatting on output

that saves a little space in storage, and would make it easier for rgw internals to compare/manipulate it with std::chrono